### PR TITLE
Gh6945 2 prefer delivery days for any partner but closer to the promised date.

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/6945

### CODE REVIEW

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/tourplanning/api/impl/DeliveryDayDAO.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
 				.addOnlyActiveRecordsFilter()
 				//
 				// for given BP & Location
-				.addInArrayFilter(I_M_DeliveryDay.COLUMNNAME_C_BPartner_ID , null, params.getBPartnerLocationId().getBpartnerId())
+				.addInArrayFilter(I_M_DeliveryDay.COLUMNNAME_C_BPartner_ID, null, params.getBPartnerLocationId().getBpartnerId())
```
Comment (teosarca) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/tourplanning/api/impl/DeliveryDayDAO.java:77]: what if `params.getBPartnerLocationId()` is null?

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/tourplanning/model/validator/C_Order.java [hunk 2]
```diff
@@ -44,6 +43,8 @@
 @Interceptor(I_C_Order.class)
 public class C_Order
 {
+	final IOrderDeliveryDayBL orderDeliveryDayBL = Services.get(IOrderDeliveryDayBL.class);
```
Comment (teosarca) [inline backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/tourplanning/model/validator/C_Order.java:46]: make it private

File: backend/de.metas.util/src/test/java/de/metas/util/time/SystemTimeTest.java [hunk 3]
```diff
@@ -80,5 +84,23 @@ public void test_asZonedDateTimeAtStartOfDay()
 		assertThat(dayTS.getSecond()).isEqualTo(0);
 		assertThat(dayTS.getNano()).isEqualTo(0);
 	}
+	
+	@Test
+	public void test_2359()
```
Comment (teosarca) [inline backend/de.metas.util/src/test/java/de/metas/util/time/SystemTimeTest.java:89]: what's the purpose of this test? (i.e. what it tests)
what's the relation with SystemTime class?

File: backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/tourplanning/api/impl/DeliveryDayDAO.java (hunk 1 above)
Comment (metas-rc) [reply backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/tourplanning/api/impl/DeliveryDayDAO.java:77]: true. Fixed

File: backend/de.metas.util/src/test/java/de/metas/util/time/SystemTimeTest.java (hunk 3 above)
Comment (metas-rc) [reply backend/de.metas.util/src/test/java/de/metas/util/time/SystemTimeTest.java:89]: As per my concept, I wanted to make sure that particular line
```java
final ZonedDateTime todayEndOfDay = SystemTime.asLocalDate().atTime(LocalTime.MAX).atZone(SystemTime.zoneId());
```
is retrieving the date with hour 23:59. 
True, it doesn't actually test SystemTime. I'll just take it out. It's not such a relevant test. I suppose LocalTime.MAX won't change over time.

**Pull request merged**